core/src/apps/common/request_pin.py

```diff
@@ -124,18 +124,28 @@ async def verify_user_pin(
 
 
 async def error_pin_invalid(ctx: wire.Context) -> NoReturn:
-    from apps.common.confirm import confirm
-
-    text = Text("Wrong PIN", ui.ICON_WRONG, ui.RED)
-    text.normal("The PIN you entered is", "invalid.")
-    await confirm(ctx, text, confirm=None, cancel="Close")
+    from trezor.ui.layouts import show_error_and_raise
+
+    await show_error_and_raise(
+        ctx,
+        "warning_wrong_pin",
+        header="Wrong PIN",
+        content="The PIN you entered is invalid.",
+        red=True,
+        exc=wire.PinInvalid,
+    )
     raise wire.PinInvalid
 
 
 async def error_pin_matches_wipe_code(ctx: wire.Context) -> NoReturn:
-    from apps.common.confirm import confirm
-
-    text = Text("Invalid PIN", ui.ICON_WRONG, ui.RED)
-    text.normal("The new PIN must be", "different from your", "wipe code.")
-    await confirm(ctx, text, confirm=None, cancel="Close")
+    from trezor.ui.layouts import show_error_and_raise
+
+    await show_error_and_raise(
+        ctx,
+        "warning_invalid_new_pin",
+        header="Invalid PIN",
+        content="The new PIN must be different from your\nwipe code.",
+        red=True,
+        exc=wire.PinInvalid,
+    )
     raise wire.PinInvalid
```

Comment:
isn't this dead code now?

Reply:
Fixed in fef81373696d5dc2929b40dba0343afe855c9a7f.